### DISCUSSION TITLE
Alarms with sound

### DISCUSSION THREAD

**Discussion by robsori** [frangoteam/FUXA#1142] [Ideas]
I have met SCADA projects in which it was requested that certain critical alarms be highlighted by a sound. I think it would be useful to be able to set a sound alert for the desired event in Fuxa, in the alarm configuration page.

**Comment by unocelli** [comment]
Hi, in scada acoustic alarms are handled via output tags

**Reply by henjoe** [reply]
Is it already available in Fuxa?

**Reply by unocelli** [reply]
Hi, there doesn't seem to be a library, you can make a script that launches a python script.
```
import sys

try:
    import winsound
    def beep():
        winsound.Beep(1000, 500)

beep()
```

**Reply by henjoe** [reply]
Is it possible to run python scripts in Fuxa like that?? 👀

**Reply by unocelli** [reply]
Yes, I tested in windows

**Reply by henjoe** [reply]
How can I do the pip install in Fuxa? Like for example I wanna use specific library.

**Reply by unocelli** [reply]
I don't know...I asked chatgpt 😉

**Reply by henjoe** [reply]
TBH, I dont know if you are being sarcastic 😊 
As far as I know, Fuxa script's backend is nodejs,  how can I use external python scripts on it? 
Assuming you're just being sarcastic, I would be interested to use nodejs libraries, for example "axios" to use it Fuxa in scripts to send HTTP request, will it be possible? Can I just simply 
var ax = require('axios') inside the Fuxa scripts?

**Reply by IonuDragomir1** [reply]
Hello,
I have tested @unocelli's code but doesn't work. Am I missing something? Do I have to install a module in order to run that code? I have installed python on my windows OS.

**Reply by henjoe** [reply]
It will never work, fuxa scripts is based on nodejs, the example of @unocelli  is based on python. 

It will be a highly appteciated my friend @unocelli gives us instruction how your example python code will work on fuxa scripts?

**Reply by unocelli** [reply]
![image](https://github.com/frangoteam/FUXA/assets/48684550/1d3a9d0a-7109-4940-aadd-42e17c0a3cc9)

**Reply by henjoe** [reply]
Thanks @unocelli  now I get it.

**Reply by henjoe** [reply]
One last thing mt friend @unocelli , how can I apply the scripts everytime there is an alarm event? Do we have a global event that we can monitor and bind that event to our sound alarm script? 
What I wanna do is simply everytime there is an alarm (could HH, LL, PH and etc ), I wanna trigger our soubd alarm script to inforn the operator. 

By the way I wanna try and test this nodejs library
https://www.npmjs.com/package/sound-play

If it works good, I'll tryna create a PR.

Regards,
Henjoe

**Reply by IonuDragomir1** [reply]
For me it doesn't work yet.
My phyton beep.py code:
```
>>> import winsound
>>> frequency = 5000
>>> duration = 1000
>>> winsound.Beep(frequency, duration)
```
Each time I call `winsound.Beep(frequency, duration)`, it works as expected. I get the sound. 
I created a script on fuxa, as @unocelli suggested:

```
const {spawn} = require('child_process');
const path = require('path');
console.log("path: " + path);


const scriptPath = path.join('C:\\Users\\MyUsr\\Desktop', 'beep.py');
console.log("scriptPath: " + scriptPath);

const pythonProcess = spawn('python', [scriptPath]);
```
If I call this script on a button, I don't get any sound. What am I doing wrong?

**Reply by henjoe** [reply]
Maybe @unocelli  can help you with that. 
Does the sound permission on your browser is enabled also?

**Reply by IonuDragomir1** [reply]
Yes, sound is enabled for my browser.

**Reply by henjoe** [reply]
> One last thing mt friend @unocelli , how can I apply the scripts everytime there is an alarm event? Do we have a global event that we can monitor and bind that event to our sound alarm script? 
> What I wanna do is simply everytime there is an alarm (could HH, LL, PH and etc ), I wanna trigger our soubd alarm script to inforn the operator. 
> 
> By the way I wanna try and test this nodejs library
> https://www.npmjs.com/package/sound-play
> 
> If it works good, I'll tryna create a PR.
> 
> Regards,
> Henjoe

Hi @unocelli how will I know if there is an alarm existing, is there a global variable to trigger that? I am halfway working on this sound, and I wanna bind the "sound script" if there is a New Alarm. 
Hope to help me here. 

Regards,

**Reply by unocelli** [reply]
Hi, we need $getAlarms system function to use in script

**Reply by ARSuleimanov** [reply]
Hi! how do I get the number of unacknowledge alarms using $getAlarms function?
